source/NVDAObjects/IAccessible/ia2Web.py

```diff
@@ -213,8 +213,40 @@ def _get_states(self):
 			if popupState:
 				states.discard(controlTypes.State.HASPOPUP)
 				states.add(popupState)
+		if self.isInternalLink:
+			states.add(controlTypes.State.INTERNAL_LINK)
 		return states
 
+	@property
+	def isInternalLink(self) -> bool:
+		if self.role != controlTypes.Role.LINK:
+			return False
+		value = self.value
+		if not value:
+			return False
+		if not hasattr(self, "treeInterceptor"):
+			return False
+		ti = self.treeInterceptor
+		if ti is None or not hasattr(ti, "documentConstantIdentifier"):
+			return False
+		documentConstantIdentifier = ti.documentConstantIdentifier
+		if self._valueToSamePage(value, documentConstantIdentifier):
+			return True
+		return False
+
+	def _valueToSamePage(self, value: str, constantIdentifier: str) -> bool:
```

Comment:
Could you make this a global utility? May be we need a module for this, e.g. webUtils or urlUtils.
You could also consider using `urllib.parse` to get the constant part of the url.

Reply:
I prefer urlUtils than webUtils. Seems a more precise name.

Reply:
Done, for now without using urllib, just with a very simple private function.